components/timezone/src/iana_ids.rs

```diff
@@ -0,0 +1,243 @@
+    data1: DataPayload<IanaToBcp47MapV1Marker>,
+    data2: DataPayload<Bcp47ToIanaMapV1Marker>,
+}
+
+impl IanaBcp47RoundTripMapper {
+    /// Creates a new [`IanaBcp47RoundTripMapper`].
+    ///
+    /// See [`IanaBcp47RoundTripMapper`] for an example.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    #[inline]
+    pub const fn new() -> Self {
+        IanaBcp47RoundTripMapper {
+            data1: DataPayload::from_static_ref(
+                crate::provider::Baked::SINGLETON_TIME_ZONE_IANA_TO_BCP47_V1,
+            ),
+            data2: DataPayload::from_static_ref(
+                crate::provider::Baked::SINGLETON_TIME_ZONE_BCP47_TO_IANA_V1,
+            ),
+        }
+    }
+
+    icu_provider::gen_any_buffer_data_constructors!(locale: skip, options: skip, error: TimeZoneError,
+        #[cfg(skip)]
+        functions: [
+            new,
+            try_new_with_any_provider,
+            try_new_with_buffer_provider,
+            try_new_unstable,
+            Self,
+        ]
+    );
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new)]
+    pub fn try_new_unstable<P>(provider: &P) -> Result<Self, TimeZoneError>
+    where
+        P: DataProvider<IanaToBcp47MapV1Marker> + DataProvider<Bcp47ToIanaMapV1Marker> + ?Sized,
+    {
+        let data1 = provider.load(Default::default())?.take_payload()?;
+        let data2 = provider.load(Default::default())?.take_payload()?;
+        let obj = Self { data1, data2 };
+        if obj.data1.get().bcp47_ids_checksum != obj.data2.get().bcp47_ids_checksum {
+            return Err(TimeZoneError::MismatchedChecksums);
+        }
+        Ok(obj)
+    }
+
+    /// Returns a borrowed version of the mapper that can be queried.
+    ///
+    /// This avoids a small potential cost of reading the data pointer.
+    pub fn as_borrowed(&self) -> IanaBcp47RoundTripMapperBorrowed {
+        IanaBcp47RoundTripMapperBorrowed {
+            data1: self.data1.get(),
+            data2: self.data2.get(),
+        }
+    }
+}
+
+/// A borrowed wrapper around IANA-BCP47 time zone data, returned by
+/// [`IanaBcp47RoundTripMapper::as_borrowed()`]. More efficient to query.
+#[derive(Debug)]
+pub struct IanaBcp47RoundTripMapperBorrowed<'a> {
+    data1: &'a IanaToBcp47MapV1<'a>,
+    data2: &'a Bcp47ToIanaMapV1<'a>,
+}
+
+impl<'a> IanaBcp47RoundTripMapperBorrowed<'a> {
+    /// Looks up a BCP-47 time zone identifier based on an ASCII-case-insensitive match for
+    /// the given IANA time zone identifier.
+    ///
+    /// This is the type of match specified in [ECMAScript Temporal].
+    ///
+    /// See examples in [`IanaToBcp47Mapper`].
+    ///
+    /// [ECMAScript Temporal]: https://tc39.es/proposal-temporal/#sec-isavailabletimezonename
+    pub fn iana_to_bcp47(&self, iana_id: &str) -> Option<TimeZoneBcp47Id> {
```

Comment:
here it's called `iana_to_bcp47`, whereas on the other mapper it's just `get`. It'd be cool if there was only one type with two constructors and some generic shenanigans that enables the second method.

Reply:
Acknowledged and happy to bikeshed if you have suggestions